### ISSUE TITLE
Trouble W/ Functional Migrations within a Docker Container

### ISSUE THREAD

**Issue by TDS-git** [mikro-orm/mikro-orm#1494]
**Describe the bug**
I've been trying to automate the setup of a Postgresql, express, and docker host and I am one step away from completing the goal, but this error/bug/lack of understanding is starting to make me question whether I'm actually doing this wrong or I'm crazy.

The problem - 
Using orm.getMigrator().up(), to create a table within a new database created with docker-compose does not seem to run the up() function within the Migration object. Which doesn't create any tables or actually run any of the migrations...

config object { ... }  as Parameters<\'typeof MikroORM.init\'>[0] (Without the quotes)
This happens only when I am passing the MikroORM.init() method the config object above that uses the Path + Pattern approach to detect the location of Migrations that the database needs to create tables upon instantiation.

If I use the migration list property within my config object to supply static migrations it finds the migrations and will create the table, allowing the persistAndFlush to run properly (Woo hoo, this also only works if the path and pattern have been included in the config object). 

Other than that I have had no luck trying to get the migration to work within a docker-container using just the path and pattern approach. This worries me that I am going to have to supply a static migration to the migrationList property every time to make this work? (It doesn't appear to be that way in Ben Awad's videos)

**I am also aware that I am connecting to database postgres and not tvtrader. For some reason docker-compose is also not creating the data base with the name specified by the environment variable. So I just used the default postgres database that's created everytime.

**Stack traceS**
Running in containers w/ createMigration function

![Using_create_migration](https://user-images.githubusercontent.com/71679719/109209170-10f52880-7760-11eb-9017-ee3c73733a0c.png)

Running in containers w/ getMigrator().up() function

![Using_only_getMigrator_up](https://user-images.githubusercontent.com/71679719/109209233-2407f880-7760-11eb-9c68-0a852b91f0c9.png)

Running in container using static migration path 
![Success-within-container](https://user-images.githubusercontent.com/71679719/109210977-721dfb80-7762-11eb-98ce-98831bc3bc3b.png)

Running locally w only path + pattern w/ getMigrator().up() function
![Running_Locally](https://user-images.githubusercontent.com/71679719/109209497-7812dd00-7760-11eb-843b-d5a2b2b086b2.png)

Other files--------------------------------

Container Setup in Docker Desktop
![Container_Setup](https://user-images.githubusercontent.com/71679719/109209369-51ed3d00-7760-11eb-93be-e88e305dc72f.png)

Docker File
![DockerFile](https://user-images.githubusercontent.com/71679719/109209791-d049df00-7760-11eb-8f47-81ab5231781f.png)

Docker Compose
![Docker-compose](https://user-images.githubusercontent.com/71679719/109209565-8a8d1680-7760-11eb-8a58-c5dc1bc00795.png)

TS Config
![TSConfig](https://user-images.githubusercontent.com/71679719/109209413-60d3ef80-7760-11eb-98ea-c3977d54cc84.png)

ORM Config Object - WHEN LOCAL
![MikroOrmSetup](https://user-images.githubusercontent.com/71679719/109209445-66c9d080-7760-11eb-81c5-95beffdb12eb.png)

ORM Config Object - WHEN IN CONTAINER
![MikroOrmSetup_inContainer](https://user-images.githubusercontent.com/71679719/109210556-d8564e80-7761-11eb-9e3f-e33989c2e7b9.png)

ORM Config Object - WHEN IN CONTAINER W/ SUCCESSFUL MIGRATION
![MikroOrmSetup_inContainer_success](https://user-images.githubusercontent.com/71679719/109210928-5f0b2b80-7762-11eb-9a64-b4ec58c88d55.png)

Package.json
![package_json](https://user-images.githubusercontent.com/71679719/109209466-6df0de80-7760-11eb-8e7e-d10b99a0b646.png)

Main Method
![main_method](https://user-images.githubusercontent.com/71679719/109209536-82cd7200-7760-11eb-8b34-642daffd3b55.png)

**To Reproduce**
1. Try to run the getMigrator().up() from a separate container using path + pattern config to find migration files.

**Expected behavior**
Trying to Create a table

This is my first time submitting an issue to a github repo, please go easy on me xD. If this is not the right place to submit this I will move it ASAP.

**Comment by B4nan** [comment]
Ouch, this is quite challenging to review, next time create a repo instead of 10 screenshots :P 

First of all, this craziness is not needed at all:

`config object { ... } as Parameters<'typeof MikroORM.init'>[0] (Without the quotes)`

do this instead:

```ts
import { Options } from '@mikro-orm/core'; // `Options` is what you want, no need to get the type from init method

const config: Options = { ... };

export default config;
```

Second, in your docker container you are executing the compiled code - there is no support for TS files this way, so that is why you saw the "Cannot use import statement..." error, as you try to read TS via node. 

There are several ways how to go about this, but in general - you need multiple configurations. For the production build (so when you run compiled code via `node`), you need to set `pattern` to read JS files only, and adjust the `path` to dist. and of course, you need to compile the migration files too.

Or you can execute migrations via `ts-node` - or via CLI that will use `ts-node`. Your CLI config should keep the `pattern` set for TS files and `path` to the src, but when you initialize the ORM in app, you can adjust the configuration as noted in previous paragraph.

Or you can just emit JS files directly and forget about all of this :]

**Comment by B4nan** [comment]
Nope, `useTsNode` is only for CLI. You don't want to use `ts-node` for production build, that would probably have performance implications.

**Comment by TDS-git** [comment]
I see, I will have to do more research on the config files, thanks for the speedy input! I appreciate your time

**Comment by B4nan** [comment]
In v4.5 it will be possible to override the configuration via env variables, so you could also use that inside the docker container.

Already available as `4.4.5-dev.20`: 

https://mikro-orm.io/docs/next/configuration/#using-environment-variables

**Comment by TDS-git** [comment]
https://github.com/TDS-git/server

I actually got it working fairly fluently, I have a repo up right now under the alias 'server' if you want to check it out and see what I did, I'm not a high-level programmer so any input on what I should change will be appreciated 😄 

Also, thank you for making the process easier for the new guys!

I'm going to follow through with the rest of the tutorial now
